kvm-ioctls/src/ioctls/vm.rs

```diff
@@ -1300,7 +1300,7 @@ impl VmFd {
     /// ```
     pub fn create_device(&self, device: &mut kvm_create_device) -> Result<DeviceFd> {
         // SAFETY: Safe because we are calling this with the VM fd and we trust the kernel.
-        let ret = unsafe { ioctl_with_ref(self, KVM_CREATE_DEVICE(), device) };
+        let ret = unsafe { ioctl_with_mut_ref(self, KVM_CREATE_DEVICE(), device) };
```

Comment:
Can you add a test that would trigger the error? As we discussed offline this would not be triggering in the CI because we're not using the latest Rust version. It's enough if we can trigger it locally because eventually we'll be using the latest version in the CI as well.

Reply:
We already have tests which can trigger this behavior. Tests are: `test_register_unregister_irqfd` and `test_set_irq_line`.